File: Code/Websites/DanpheEMR/wwwroot/DanpheApp/src/app/settings-new/billing/shared/dto/billing-scheme.dto.ts
import { BillingSubScheme_DTO } from "../../../../billing/shared/dto/bill-subscheme.dto";
import { ENUM_BillPaymentMode } from "../../../../shared/shared-enums";

export class BillingScheme_DTO {
  public SchemeId: number = 0;
  public SchemeCode: string = null;
  public SchemeName: string = "";

  //sud:10Mar'23--This property is only there in Client side for now. We may have to add this to DbTable and so on..
  public MembershipDisplayName: string = null;
  public CommunityName: string = null;
  public IsDiscountApplicable: boolean = false;
  public DiscountPercent: number = 0;
  public IsDiscountEditable: boolean = false;
  public IsMembershipApplicable: boolean = false;
  public IsMemberNumberCompulsory: boolean = false;
  public DefaultPaymentMode: string = ENUM_BillPaymentMode.cash;
  public IsCreditApplicable: boolean = false;
  public IsCreditOnlyScheme: boolean = false;
  public CreditLimit: number = 0;
  public DefaultCreditOrganizationId: number = 0;
  public IsCoPayment: boolean = false;

  public DefaultPriceCategoryId: number = 0;
  public DefaultPriceCategoryName: string = null;

  public ApiIntegrationName: string = null;
  public FieldSettingParamName: string = null;// { get; set; }
  public IsSystemDefault: boolean = false;

  public CoPaymentCashPercent: number = 0;
  public CoPaymentCreditPercent: number = 0;
  public IsCreditLimited: boolean = false;
  public IsGeneralCreditLimited: boolean = false;
  public GeneralCreditLimit: number = 0;
  public IsClaimCodeAutoGenerate: boolean = false;
  public HasSubScheme: boolean = false;
  public AllowProvisionalBilling: boolean = false;
  public SubSchemes = new Array<BillingSubScheme_DTO>();
}

File: Code/Websites/DanpheEMR/Services/Billing/DTO/BillingScheme_DTO.cs
using System.Collections.Generic;

namespace DanpheEMR.Services.Billing.DTO
{
    public class BillingScheme_DTO
    {
        public int SchemeId { get; set; }
        public string SchemeCode { get; set; }
        public string SchemeName { get; set; }
        public string CommunityName { get; set; }
        public bool IsDiscountApplicable { get; set; }
        public decimal DiscountPercent { get; set; }
        public bool IsDiscountEditable { get; set; }
        public bool IsMembershipApplicable { get; set; }
        public bool IsMemberNumberCompulsory { get; set; }
        public string DefaultPaymentMode { get; set; }
        public bool IsCreditApplicable { get; set; }
        public bool IsCreditOnlyScheme { get; set; }
        public decimal CreditLimit { get; set; }
        public int? DefaultCreditOrganizationId { get; set; }
        public bool IsCoPayment { get; set; }

        public int DefaultPriceCategoryId { get; set; }
        public string DefaultPriceCategoryName { get; set; }

        public string ApiIntegrationName { get; set; }
        public string FieldSettingParamName { get; set; }
        public decimal CoPaymentCashPercent { get; set; }
        public decimal CoPaymentCreditPercent { get; set; }
        public bool IsSystemDefault { get; set; }
        public bool IsCreditLimited { get; set; }
        public bool IsGeneralCreditLimited { get; set; }
        public decimal GeneralCreditLimit { get; set; }
        public bool IsClaimCodeAutoGenerate { get; set; }
        public bool HasSubScheme { get; set; }
        public List<BillingSubScheme_DTO> SubSchemes { get; set; }
        public bool AllowProvisionalBilling { get; set; }

    }
}
